### ISSUE TITLE
DOC: HTMLSemanticPreservingSplitter is documented in API References but not exported in langchain-text-splitters v0.3.4

### ISSUE THREAD

**Issue by aliemir** [langchain-ai/langchain#28917]
### URL

https://python.langchain.com/api_reference/text_splitters/html/langchain_text_splitters.html.HTMLSemanticPreservingSplitter.html#langchain_text_splitters.html.HTMLSemanticPreservingSplitter

### Checklist

- [X] I added a very descriptive title to this issue.
- [X] I included a link to the documentation page I am referring to (if applicable).

### Issue with current documentation:

API References documentation shows HTMLSemanticPreservingSplitter as available in langchain-text-splitters, but it's not exported in version 0.3.4. This causes import failures when trying to use the documented functionality.

Steps to reproduce:
- `pip install "langchain-text-splitters==0.3.4"`
- Check the contents of the package
- Verify that `HTMLSemanticPreservingSplitter` is not available in 0.3.4

**Comment by mohammedsalah-ai** [comment]
Yeah I have faced the same problem as well. to experiment with it I just copied the source into my file
